addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java

```diff
@@ -0,0 +1,172 @@
+                        logger.debug("Changing state to: {}", command);
+                        // Open WebSocket
+                        valloxSendSocket.request(channelUID, command.toString());
+                        valloxSendSocket.request(null, null);
+                    }
+                } catch (NumberFormatException nfe) {
+                    // Other commands like refresh
+                    return;
+                }
+            } else if (ValloxMVBindingConstants.CHANNEL_ONOFF.equals(channelUID.getId())) {
+                if (OnOffType.ON.equals(command)) {
+                    valloxSendSocket.request(channelUID, "0");
+                } else if (OnOffType.OFF.equals(command)) {
+                    valloxSendSocket.request(channelUID, "5");
+                }
+            } else if (ValloxMVBindingConstants.CHANNEL_EXTR_FAN_BALANCE_BASE.equals(channelUID.getId())) {
+                QuantityType<Dimensionless> quantity = commandToQuantityType(command, SmartHomeUnits.PERCENT);
+                valloxSendSocket.request(channelUID, Integer.toString(quantity.intValue()));
+            } else if (ValloxMVBindingConstants.CHANNEL_SUPP_FAN_BALANCE_BASE.equals(channelUID.getId())) {
+                QuantityType<Dimensionless> quantity = commandToQuantityType(command, SmartHomeUnits.PERCENT);
+                valloxSendSocket.request(channelUID, Integer.toString(quantity.intValue()));
+            } else if (ValloxMVBindingConstants.CHANNEL_HOME_SPEED_SETTING.equals(channelUID.getId())) {
+                QuantityType<Dimensionless> quantity = commandToQuantityType(command, SmartHomeUnits.PERCENT);
+                valloxSendSocket.request(channelUID, Integer.toString(quantity.intValue()));
+            } else if (ValloxMVBindingConstants.CHANNEL_HOME_AIR_TEMP_TARGET.equals(channelUID.getId())) {
+                // Convert temperatur to milli degree Kelvin
+                QuantityType<Temperature> quantity = commandToQuantityType(command, SIUnits.CELSIUS)
+                        .toUnit(SmartHomeUnits.KELVIN);
+                @SuppressWarnings("null")
+                int milliKelvin = quantity.multiply(new BigDecimal(100)).intValue();
+                valloxSendSocket.request(channelUID, Integer.toString(milliKelvin));
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    protected <U extends Quantity<U>> QuantityType<U> commandToQuantityType(Command command, Unit<U> defaultUnit) {
+        if (command instanceof QuantityType) {
+            return (QuantityType<U>) command;
+        }
+        return new QuantityType<U>(new BigDecimal(command.toString()), defaultUnit);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+
+        String ip = getConfigAs(ValloxMVConfig.class).getIp();
+        valloxSendSocket = new ValloxMVWebSocket(ValloxMVHandler.this, ip);
+
+        BigDecimal bdUpdateInterval = getConfigAs(ValloxMVConfig.class).getUpdateinterval();
```

Comment:
IMHO BigDecimal's are not the proper type for such things / calculations. You don't need decimals and you don't need big.